python/timing/utils.py

```diff
@@ -32,6 +32,7 @@ def get_model_bytes(model) -> str:
 def timing(f):
     @wraps(f)
     def wrap(*args, **kw):
+        torch.cuda.synchronize()
```

Comment:
The bug has to do with this line. 

Without this line, if you don't apply optimizer.step, the result is still correct. However, if you apply optimizer.step, the result is wrong. 

I am not sure why optimizer.step can cause this issue though. Will need to investigate it later.

Reply:
BTW, it would be better if we eliminate the timing results for experiments in which dimxdim < mem_size. In these cases, we actually enlarge the parameter size.

Reply:
Ideally, we should keep them as we plan to share memory across matrices (and other modules) and thus it is possible that memory footprint is larger than a particular matrix itself.